File: third-person/scenery.go
package main

// Scenery struct
type Scenery struct {
	*Thing
}

// NewTree func
func NewTree(world *World, x, y, z float32) *Scenery {
	t := &Scenery{}
	t.Thing = &Thing{}
	t.UID = "tree"
	t.NID = NextNID()
	t.World = world
	t.X = x
	t.Y = y
	t.Z = z
	t.Radius = 0.4
	t.Height = 1.0
	world.AddThing(t)
	t.BlockBorders()
	t.AddToBlocks()
	return t
}

// Damage func
func (me *Scenery) Damage(amount int) {
}

// Update func
func (me *Scenery) Update() {
}

File: public/scenery.js
class Tree extends Thing {
    constructor(world, nid, x, y, z) {
        super()
        this.World = world
        this.UID = TreeUID
        this.SID = "scenery"
        this.NID = nid
        this.Sprite = SpriteData[this.SID]["dead-tree"]
        this.X = x
        this.Y = y
        this.Z = z
        this.OldX = x
        this.OldY = y
        this.OldZ = z
        this.Radius = 0.4
        this.Height = 1.0
        this.Speed = 0.1
        this.Health = 1
        world.AddThing(this)
        this.BlockBorders()
        this.AddToBlocks()
    }
    Update() {}
    Render(timeNow, interpolation, spriteBuffer, camX, camZ, camAngle) {
        let sin = camX - this.X
        let cos = camZ - this.Z
        let length = Math.sqrt(sin * sin + cos * cos)
        sin /= length
        cos /= length
        Render3.Sprite(spriteBuffer[this.SID], this.X, this.Y, this.Z, sin, cos, this.Sprite)
    }
}
